src/main/scala/com/fulcrumgenomics/alignment/Alignment.scala

```diff
@@ -82,4 +120,55 @@ case class Alignment(query: Array[Byte],
 
   /** One based closed coordinate of the end of the alignment on the query sequence. */
   def targetEnd: Int = targetStart + cigar.lengthOnTarget - 1
+
+  /**
+    * Generates a padded text representation of the alignment for visualization. The returned
+    * sequence will consist of three lines as follows (minus the labels on the left):
+    *
+    * query : ACGTGAACTGACT-ACTGTATGCG
+    * align : |||||  |||||| ||||||||.|
+    * target: ACGTG--CTGACTGACTGTATGGG
+    *
+    * @param matchChar the character to use in the alignment line when the bases match
+    * @param mismatchChar the character to use in the alignment line when the bases mismatch
+    * @param gapChar the character to use in the alignment line when one of the sequences is gapped
+    * @return Three lines representing the alignment
+    */
+  def paddedString(matchChar: Char = '|', mismatchChar: Char = '.', gapChar: Char = ' '): Seq[String] = {
+    val buffers = Seq(new StringBuilder, new StringBuilder, new StringBuilder)
+    val Seq(queryBuffer, alignBuffer, targetBuffer) = buffers
+
+    var qOffset = queryStart - 1
+    var tOffset = targetStart - 1
+    cigar.foreach {
```

Comment:
Part of me wishes that if you could write this as `cigar.foldLeft` and not have to use `vars`, but then you'd have to carry forward five variables, which would be pretty ugly (unless it was part of a case class :P).  I find myself doing this a lot too, namely using `var`s and a `foreach` instead of `foldLeft`.  Sigh.